### PR TITLE
Initial Zephyr application & module support

### PR DESCRIPTION
**Pull request by lgirdwood**
Starting point for Zephyr RTOS integration.

### CODE REVIEW

File: src/include/sof/common.h [hunk 1]
```diff
@@ -24,6 +24,10 @@
 #include <stddef.h>
 
 /* use same syntax as Linux for simplicity */
+/* Zephyr defines this - remove local copy once Zephyr integration complete */
+#ifdef ARRAY_SIZE
+#undef ARRAY_SIZE
+#endif
 #define ARRAY_SIZE(x) (sizeof(x) / sizeof(x[0]))
```
[the remark this answers is not in the document]
Comment (dbaluta) [reply src/include/sof/common.h:31]: agree with @ClandTidy

File: src/include/ipc/channel_map.h [hunk 2]
```diff
@@ -41,7 +41,7 @@ struct sof_ipc_channel_map {
 	uint32_t ch_mask;
 	uint32_t reserved;
 	int32_t ch_coeffs[0];
-} __packed;
+} __attribute__((packed));
```
Comment (paulstelian97) [inline src/include/ipc/channel_map.h:44]: Can't we have a macro that does `#define __packed __attribute__((packed))` somewhere?

File: src/include/sof/common.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (ymdatta) [reply src/include/sof/common.h:31]: @dbaluta : Created the PR #2771.

File: src/include/ipc/channel_map.h (hunk 2 above)
Comment (lgirdwood) [reply src/include/ipc/channel_map.h:44]: @paulstelian97 we already have the macro, the problem is inconsistent usage and macro is also duplicated on Zephyr too.  We will have to live with some duplication until the port is done when we can clean up.

**Comment by lgirdwood** [comment]
CI known issues.

**Pull request merged**